src/preprocess.rkt

```diff
@@ -71,13 +58,28 @@
   (define sort-instructions
     (for/list ([component (in-list components)]
                #:when (> (length component) 1))
-      (cons 'sort component)))
+      (cons 'sort component)))      
+  (define preprocessing (append abs-instructions sort-instructions))
+  (define alternative (make-alt-preprocessing initial preprocessing))
+   ;; Absolute value should happen before sorting
+  (define simplified
+    (cons
+     ;; We excluded the first element of `initials` above so that we can add it
+     ;; here manually, but without a self-referential history.
+     alternative
+     (remove-duplicates
+      (for/list ([expression (in-list initials)])
+        (alt
+         expression
+         (list 'simplify null query #f #f)
+         (list alternative)
+         preprocessing))
+      alt-equal?)))
   (values
    (if (flag-set? 'setup 'simplify)
        simplified
-       (list (make-alt initial)))
-   ;; Absolute value should happen before sorting
-   (append abs-instructions sort-instructions)))
+       (list (make-alt-preprocessing initial preprocessing)))
+   preprocessing))
```

Comment:
Also unfortunately this will require some fixing merge conflicts now that Jackson's `preprocess-odd` has gone in.

Reply:
fixed